j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/TranslationsFileProcessor.java

```diff
@@ -97,17 +98,109 @@ public Optional<File> getTranslationsFile(List<Input> inputs, BuildLog log) {
 
                     String lang = translationbundleNode.item(0).getAttributes().getNamedItem("lang").getNodeValue();
 
-                    if (locale.equals(lang)) {
-                        return Optional.of(xtb);
+                    if (locales.contains(lang)) {
+                        if(!suitableFiles.containsKey(lang)) {
```

Comment:
i think i'd prefer a computeIfAbsent below to auto-populate this?

Reply:
sure, np